Posix API `strnlen` declaration not found in native_posix/native_sim

https://github.com/zephyrproject-rtos/zephyr/actions/runs/9638989345/job/26580638001?pr=74706

``` C
/__w/zephyr/zephyr/subsys/net/lib/http/http_server_http1.c
/__w/zephyr/zephyr/subsys/net/lib/http/http_server_http1.c: In function ‘dynamic_get_req’:
/__w/zephyr/zephyr/subsys/net/lib/http/http_server_http1.c:104:35: error: implicit declaration of function ‘strnlen’; did you mean ‘strlen’? [-Werror=implicit-function-declaration]
  104 |                                   strnlen(http_response,                \
      |                                   ^~~~~~~
/__w/zephyr/zephyr/subsys/net/lib/http/http_server_http1.c:116:15: note: in expansion of macro ‘SEND_RESPONSE’
  116 |         ret = SEND_RESPONSE(RESPONSE_TEMPLATE_CHUNKED,
      |               ^~~~~~~~~~~~~
cc1: all warnings being treated as errors
ninja: build stopped: subcommand failed.
```

I'm also checkout zephyr-3.5.0, this branch build passed, and also compare build command:

**zephyr-3.5.0**
```C
"command": "/usr/bin/gcc -DKERNEL -D_POSIX_C_SOURCE=200809 -D_XOPEN_SOURCE=600 -D_XOPEN_SOURCE_EXTENDED -DK_HEAP_MEM_POOL_SIZE=0 -D__ZEPHYR__=1 -I/home/mi/works/github/zephyrproject/zephyr/kernel/include
```
**main:**
```C
"command": "/usr/bin/gcc -DKERNEL -DK_HEAP_MEM_POOL_SIZE=0 -D__ZEPHYR__=1 -I/home/mi/works/github/zephyrproject/zephyr/kernel/include
```

--- a/subsys/net/lib/http/http_server_http1.c
+++ b/subsys/net/lib/http/http_server_http1.c
@@ -5,6 +5,9 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#undef _POSIX_C_SOURCE
+#define _POSIX_C_SOURCE 200809L /* Required for strnlen() */
+
 #include <errno.h>
 #include <stdbool.h>
 #include <stdio.h>

--- a/tests/net/lib/http_server/hpack/testcase.yaml
+++ b/tests/net/lib/http_server/hpack/testcase.yaml
@@ -9,5 +9,8 @@ common:
   integration_platforms:
     - native_sim
     - qemu_x86
+  platform_exclude:
+    - native_posix
+    - native_posix/native/64
 tests:
   net.http.server.http2_hpack: {}

